radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/ProxiedPeersView.java

```diff
@@ -0,0 +1,122 @@
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.network.p2p;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import com.radixdlt.environment.RemoteEventDispatcher;
+import com.radixdlt.network.p2p.P2PConfig.ProxyConfig;
+import com.radixdlt.network.p2p.PeerEvent.PeerConnected;
+import com.radixdlt.network.p2p.PeerEvent.PeerDisconnected;
+import com.radixdlt.network.p2p.discovery.ProxiedPeers;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public final class ProxiedPeersView implements PeersView {
+  private final PeerManager peerManager;
+  private final ProxyConfig proxyConfig;
+  private final RemoteEventDispatcher<ProxiedPeers> dispatcher;
+  private Set<RadixNodeUri> peers = Set.of();
+
+  @Inject
+  public ProxiedPeersView(
+      PeerManager peerManager,
+      ProxyConfig proxyConfig,
+      RemoteEventDispatcher<ProxiedPeers> dispatcher) {
+    this.peerManager = peerManager;
+    this.proxyConfig = proxyConfig;
+    this.dispatcher = dispatcher;
+  }
+
+  public void proxiedPeersEventProcessor(ProxiedPeers proxiedPeers) {
+    this.peers = proxiedPeers.peers();
+  }
+
+  public void peerEventProcessor(PeerEvent peerEvent) {
+    if (peerEvent instanceof PeerConnected || peerEvent instanceof PeerDisconnected) {
+      distributePeerInfo();
+    }
+  }
+
+  private void distributePeerInfo() {
+    var activePeers = new ProxiedPeers(peerManager.activePeers());
+    var proxiedPeers =
+        peerManager.activePeers().stream()
+            .filter(this::isAuthorizedProxyNode)
+            .map(uri -> uri.getNodeId().asBFTNode())
+            .toList();
+    dispatcher.dispatch(proxiedPeers, activePeers);
+  }
+
+  private boolean isAuthorizedProxyNode(RadixNodeUri uri) {
```

Comment:
I know the naming is a bit confusing, but let's stick to "proxyNode" for the node that forwards the messages (sentry node) and "proxiedNode" for the node that uses the proxy (validator).
So this should be: `isAuthorizedProxiedNode`

Reply:
Done